### PR TITLE
Homework_3

### PR DESCRIPTION
**Pull request by parfenovaaa**

### CODE REVIEW

**Review by akulebyakin — CHANGES_REQUESTED** [review]
Changes required. Please review comments.

File: src/test/java/com/epam/tc/hw3/TestData.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package com.epam.tc.hw3;
+
+public class TestData {
+
+    public final String title = "Home Page";
+    public final String userName = "ROMAN IOVLEV";
```
[the remark this answers is not in the document]
Comment (parfenovaaa) [reply src/test/java/com/epam/tc/hw3/TestData.java]: moved userName to properties file

File: src/test/java/com/epam/tc/hw3/components/LoginMenu.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    public void getProp() throws IOException {
+        FileInputStream file;
+        Properties property = new Properties();
+
+        file = new FileInputStream("src/test/resources/config.properties");
+        property.load(file);
+
+        login = property.getProperty("login.name");
+        password = property.getProperty("login.password");
+    }
```
[the remark this answers is not in the document]
Comment (parfenovaaa) [reply src/test/java/com/epam/tc/hw3/components/LoginMenu.java]: GetProp() moves to new class GetProperties

File: src/test/java/com/epam/tc/hw3/components/ElementsPage.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+        waterCheckBox.click();
+        windCheckBox.click();
+        earthCheckBox.click();
+        fireCheckBox.click();
+
+        return logScreen.getText()
+                  .replaceAll(hourRegExp, "")
+                  .split("\n");
+    }
```
[the remark this answers is not in the document]
Comment (parfenovaaa) [reply src/test/java/com/epam/tc/hw3/components/ElementsPage.java]: Method separated to clickOn() and getLog() methods

File: src/test/java/com/epam/tc/hw3/components/ElementsPage.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        dropdown = new Select(select);
+        dropdown.selectByVisibleText("Green");
+        dropdown.selectByVisibleText("Yellow");
+        dropdown.selectByVisibleText("Red");
+        dropdown.selectByVisibleText("Blue");
+
+        return logScreen.getText()
+                  .replaceAll(hourRegExp, "")
+                  .split("\n");
+    }
```
[the remark this answers is not in the document]
Comment (parfenovaaa) [reply src/test/java/com/epam/tc/hw3/components/ElementsPage.java]: ok

File: src/test/java/com/epam/tc/hw3/components/ElementsPage.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        return selenRadio.isSelected();
+    }
+
+    public boolean selectYellowDropdown() {
+        webDriver.navigate().to(webDriver.getCurrentUrl());
+        dropdown = new Select(select);
+        dropdown.selectByVisibleText("Yellow");
+        return logScreen.getText().replaceAll(hourRegExp, "")
+                          .equals(" Colors: value changed to Yellow");
+    }
```
[the remark this answers is not in the document]
Comment (parfenovaaa) [reply src/test/java/com/epam/tc/hw3/components/ElementsPage.java]: Method changed to selectDropdownByVisibleText(String colour) method

File: src/test/java/com/epam/tc/hw3/components/ElementsPage.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    public ElementsPage(WebDriver driver) {
+        this.webDriver = driver;
+        PageFactory.initElements(driver, this);
+    }
+
+    public boolean selectWaterWindCheckBox() {
+        waterCheckBox.click();
+        windCheckBox.click();
+        return (waterCheckBox.isSelected() && windCheckBox.isSelected());
+    }
```
[the remark this answers is not in the document]
Comment (parfenovaaa) [reply src/test/java/com/epam/tc/hw3/components/ElementsPage.java]: changed the logic of using ElementPage class

File: src/test/java/com/epam/tc/hw3/components/ElementsPage.java [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.Select;
+
+public class ElementsPage {
+
+    WebDriver webDriver;
+
+    //CheckBox Select
+    @FindBy(xpath = "//label[contains(., 'Water')]/input")
+    WebElement waterCheckBox;
```
[the remark this answers is not in the document]
Comment (parfenovaaa) [reply src/test/java/com/epam/tc/hw3/components/ElementsPage.java]: ok

File: src/test/java/com/epam/tc/hw3/components/ElementsPage.java [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.Select;
+
+public class ElementsPage {
+
+    WebDriver webDriver;
```
[the remark this answers is not in the document]
Comment (parfenovaaa) [reply src/test/java/com/epam/tc/hw3/components/ElementsPage.java]: guess it`s not necessary to use it in this class and in LoginMenu class, right?

File: src/test/java/com/epam/tc/hw3/ex1/Exercise1.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+import com.epam.tc.hw3.components.IndexPage;
+import com.epam.tc.hw3.components.LoginMenu;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class Exercise1 extends BaseTest {
+
+    @Test
+    public void exerciseOneTest() {
+        TestData testData = new TestData();
```
[the remark this answers is not in the document]
Comment (parfenovaaa) [reply src/test/java/com/epam/tc/hw3/ex1/Exercise1.java]: ok

File: src/test/java/com/epam/tc/hw3/TestData.java [hunk 10]
```diff
@@ -0,0 +1,34 @@
+package com.epam.tc.hw3;
+
+public class TestData {
+
+    public final String title = "Home Page";
```
[the remark this answers is not in the document]
Comment (parfenovaaa) [reply src/test/java/com/epam/tc/hw3/TestData.java]: ok

File: src/test/resources/com/epam/tc/hw3/Homework3TestSuite.xml [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+  <test name="PageObject">
+    <classes>
+      <class name="com.epam.tc.hw3.ex1.Exercise1"/>
+      <class name="com.epam.tc.hw3.ex2.Exercise2"/>
+      <class name="com.epam.tc.hw3.components.IndexPage"/>
+      <class name="com.epam.tc.hw3.components.ElementsPage"/>
+      <class name="com.epam.tc.hw3.components.LoginMenu"/>
+      <class name="com.epam.tc.hw3.BaseTest"/>
+    </classes>
+  </test>
```
[the remark this answers is not in the document]
Comment (parfenovaaa) [reply src/test/resources/com/epam/tc/hw3/Homework3TestSuite.xml]: ok, got it. Just wasn`t sure how to name test here)

**Review by akulebyakin — APPROVED** [review]
Approved!

**Pull request merged**